clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/model/EcsTask.java

```diff
@@ -82,6 +84,10 @@ private HealthState calculateHealthState(String lastStatus, String desiredStatus
       currentState = HealthState.Unknown;
     }
 
+    if ("UNHEALTHY".equals(healthStatus)) {
+      currentState = HealthState.Down;
+    }
+
```

Comment:
Is this method supposed to be a dupe of the logic in ContainerInformationService?  AFAICT this health state is used only in the UI, but hopefully should match the task state that the system is actually using to determine whether the deployment is done.  So for example, a task with its defined health check in "unknown" state should show up in the UI as "Starting", but based on this logic it could show up as "Up".

Reply:
Currently the logic differs between what the ECSTask uses to calculateHealthState and what the ContainerInformationService uses to determine platformHealthStatus. Do we want to make them the same?

Reply:
> AFAICT this health state is used only in the UI

In pipelines/during deployment, or everywhere instance (task) health is shown, such as the cluster page? 
If the latter, do we know whether changing the verbiage here will result in broken or less descriptive displays in `deck`? UI components may be looking for specific values re: which icons to show, etc.

Reply:
Hmm if this method is returning what the pipeline UI is showing during a deployment, then the risk of not making them exactly the same is that users could see "5/5 tasks up" on a deployment that which is still pending, and could eventually fail, which would be misleading.

What would it take to be able to determine here whether or not a `RUNNING` task has a healthcheck or not, so we might fork the logic in a similar way to ContainerInformationService?

Reply:
When the task is constructed, we can add a property that stores whether a health check si defined and use that to determine the health status after the fact.

Reply:
Updated the logic, in the process of adding more unit tests

Reply:
Tests added